### PR TITLE
Update the usage of the transform dialect interpreter

### PR DESCRIPTION
**Pull request by nicolasvasilache**
Upstream is shifting to using the simpler form of the transform dialect interpreter introduced in https://github.com/llvm/llvm-project/pull/68661

Update IREE's codegen usage of the interpreter as well as tests which now require a named sequence entry point.

### CODE REVIEW

File: compiler/src/iree/compiler/Codegen/Common/TransformDialectInterpreterPass.cpp
Comment (qedawkins) [inline compiler/src/iree/compiler/Codegen/Common/TransformDialectInterpreterPass.cpp:57]: nit: I would add an assert here for successfully finding the entry point. Helps for when users try to invoke this pass in isolation.

File: compiler/src/iree/compiler/Codegen/Common/MaterializeUserConfigs.cpp
Comment (qedawkins) [inline compiler/src/iree/compiler/Codegen/Common/MaterializeUserConfigs.cpp]: nit: remove commented line

File: compiler/src/iree/compiler/Codegen/Common/Passes.td
Comment (qedawkins) [inline compiler/src/iree/compiler/Codegen/Common/Passes.td]: nit: Add a description for the pass option, like
```
File path to load a library of transform dialect strategies from.
```

**Review by qedawkins — COMMENTED** [review]
Holding off on approval until all of the CI failures are unwound, but the material changes in this PR (to the interpreter/user config passes) look good to me.

Additionally I expect you'll need to remove/disable the sample here for now: https://github.com/openxla/iree/tree/main/samples/transform_dialect

File: tests/transform_dialect/cuda/softmax_partial.mlir
Comment (qedawkins) [inline tests/transform_dialect/cuda/softmax_partial.mlir]: I think your CI failures are from missing a pipe `|` here

**Review by qedawkins — APPROVED** [review]
Approving, contingent on removing the LLVM cherry pick.

File: compiler/src/iree/compiler/Codegen/Common/GPU/test/BUILD.bazel
Comment (qedawkins) [inline compiler/src/iree/compiler/Codegen/Common/GPU/test/BUILD.bazel:10]: One more question, what is the need for all of these changes to the bazel build files?

**Comment by qedawkins** [comment]
The most recent integrate should have pulled in the cherry-picked changes: https://github.com/openxla/iree/commit/18b3dd78832c4f6f4c8961901d328bca96c7efc3

File: compiler/src/iree/compiler/Codegen/Common/GPU/test/BUILD.bazel
Comment (nicolasvasilache) [reply compiler/src/iree/compiler/Codegen/Common/GPU/test/BUILD.bazel:10]: I have had this local script for a while now that I always run before committing:

``` 
iree-presubmits() {
  python ${IREE_SOURCE_DIR}/build_tools/bazel_to_cmake/bazel_to_cmake.py
  (cd ${IREE_SOURCE_DIR} && ./build_tools/scripts/run_buildifier.sh)
}
```

It picked up those changes.

**Pull request merged**
